shopgate/pwa
# Fixed the Scanner to directly open the found item, when the search result only contains one item

# Description

Fixed the Scanner to directly open the found item, when the search result only contains one of them. Refactored the Scanner actions to avoid duplicated code. Fixed/implemented unit tests.

## Type of change


## How to test it

* Run the modifications locally and check if the Scanner still works for all kinds of qr-codes and barcodes.
* It should still show the correct modals when the scan result is empty for categories, products or search queries.
* Barcodes linking to items (via id, hash or any number which would find this item) should still open the product detail page directly.
* The search qr code should skip the search results page, effectively redirecting to the product detail page, when only one item has been or can be found
* It should still show the results page when multiple items are found.
